engine/src/opensslsocket.cpp

```diff
@@ -752,6 +752,11 @@ MCSocket *MCS_open_socket(MCNameRef name, Boolean datagram, MCObject *o, MCNameR
 
 	MCSocketHandle sock = socket(AF_INET, datagram ? SOCK_DGRAM : SOCK_STREAM, 0);
 
+	// HH-2017-01-26: [[ Bug 18454 ]] Set socket option to allow broadcast
+	int t_broadcast;
+	t_broadcast = MCallowdatagrambroadcasts ? 1 : 0;
+	setsockopt(sock, SOL_SOCKET, SO_BROADCAST, (const char *)&t_broadcast, sizeof(t_broadcast));
+
```

Comment:
The `SO_BROADCAST` option should only be set if the socket is a datagram socket i.e. we should only execute these lines if `datagram` is true.